zebra-rpc/src/methods/get_block_template_rpcs/types/get_block_template/proposal.rs

```diff
@@ -42,23 +42,22 @@ pub enum ProposalResponse {
 
 impl ProposalResponse {
     /// Return a rejected response containing an error kind and detailed error info.
-    pub fn rejected<S: ToString>(kind: S, error: BoxError) -> Self {
-        let kind = kind.to_string();
-
-        // Pretty-print the detailed error for now
-        ProposalResponse::Rejected(format!("{kind}: {error:#?}"))
-    }
-
-    /// Return a rejected response containing just the detailed error information.
-    pub fn error(error: BoxError) -> Self {
-        // Pretty-print the detailed error for now
-        ProposalResponse::Rejected(format!("{error:#?}"))
-    }
-}
-
-impl<E: Error + Send + Sync + 'static> From<E> for ProposalResponse {
-    fn from(error: E) -> Self {
-        Self::error(error.into())
+    ///
+    /// Note: Error kind is currently ignored to match zcashd error format (`kebab-case` string).
+    pub fn rejected<S: ToString>(_kind: S, error: BoxError) -> Self {
```

Comment:
> Could we remove `_kind` altogether or add it to the message? (It's currently either "invalid proposal" or "invalid proposal format" so we could optionally make it an enum if we want to keep it).

I don't think it adds much to the error message, so I don't mind which way we fix this.

> Do you think we should also leave a TODO for explicitly downcasting and matching errors to constants?

This will only matter if miners care about the specific error text. I'll go ask some of the mining pools.

Reply:
(I just checked, the mining pools we're in contact with don't use block proposals currently.)